File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:

        n = len(nums)
        dp = [1] * n
        maxVal = dp[0]
        for i in range(1, n):
            for j in range(0, i):
                if nums[j] < nums[i]:
                    dp[i] = max(dp[i], dp[j] + 1)
            maxVal = max(maxVal, dp[i])

        return maxVal


        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> numSet = Arrays.stream(nums).boxed().collect(Collectors.toSet());
        int maxLen = 0;

        for (int num : nums) {
            // If num don't have left number
            // It is start of a sequence
            if (!numSet.contains(num-1)) {
                int len = 1;
                int temp = num;
                while (numSet.contains(++temp)) {
                    len++;
                }
                maxLen = Math.max(maxLen, len);
            }
        }

        return maxLen;
    }
}
